service/core.go

```diff
@@ -193,3 +198,14 @@ func (c *Core) nextTaskID() int64 {
 func (c *Core) requestLogger(r *http.Request) *zap.Logger {
 	return c.logger.With(zap.String("request_id", api.RequestIDFromContext(r.Context())))
 }
+
+func (c *Core) publishEvent(event string, data []byte) {
+	payload := fmt.Sprintf("event: %s\ndata: %s\n\n", event, string(data))
+	go func() {
+		c.subscriptionsMu.RLock()
+		for sub := range c.subscriptions {
+			sub <- []byte(payload)
```

Comment:
I could be wrong here, but I think the compiler is going to allocate a new slice on each trip through this loop, so I'd hoist the conversion to `[]byte` out of the loop (and, while you're at it, outside of the critical section).